stdlib/test/utils/test_string_slice.mojo

```diff
@@ -487,6 +487,103 @@ def test_splitlines():
         _assert_equal(s.splitlines(keepends=True), items)
 
 
+def test_split():
+    alias S = StringSlice[StaticConstantOrigin]
+
+    fn st(value: StringLiteral) -> StringSlice[ImmutableAnyOrigin]:
+        return rebind[StringSlice[ImmutableAnyOrigin]](StringSlice(value))
+
+    # FIXME: remove once StringSlice conforms to TestableCollectionElement
```

Comment:
**Question** Is there anything blocking us from doing this for `StringSlice` today?

Reply:
Didn't want to extend this PR too much. But there is probably nothing blocking. I can open another PR if I have time this weekend.